cardano-cli/src/Cardano/CLI/Byron/Key.hs

```diff
@@ -58,13 +61,13 @@ renderByronKeyFailure err =
     VerificationKeyDeserialisationFailed vKeyFp deSerError ->
       "Error deserialising verification key at: " <> textShow vKeyFp <> " Error: " <> textShow deSerError
 
-newtype NewSigningKeyFile =
-  NewSigningKeyFile FilePath
-  deriving (Eq, Ord, Show, IsString)
+newtype NewSigningKeyFile (direction :: FileDirection) = NewSigningKeyFile
+  { unNewSigningKeyFile :: File direction
+  } deriving newtype (Eq, Ord, Show, IsString, HasFileMode, MapFile)
 
-newtype NewVerificationKeyFile =
-  NewVerificationKeyFile FilePath
-   deriving (Eq, Ord, Show, IsString)
+newtype NewVerificationKeyFile (direction :: FileDirection) = NewVerificationKeyFile
+  { unNewVerificationKeyFile :: File direction
+  } deriving newtype (Eq, Ord, Show, IsString, HasFileMode, MapFile)
```

Comment:
Deriving an instance of `HasFileMode` and `MapFile` for a `newtype` is very easy.